src/main/java/org/quantumbadger/redreader/reddit/CommentListingRequest.java

```diff
@@ -296,6 +302,43 @@ private void buildCommentTree(
 		} else if(thing.getKind() == RedditThing.Kind.COMMENT) {
 
 			final RedditComment comment = thing.asComment();
+
+			if (comment.media_metadata != null && comment.media_metadata.asObject() != null) {
+				try {
+					final JsonObject metadataObj = comment.media_metadata.asObject();
+
+					final Iterator<Map.Entry<String, JsonValue>> jsonIterator = metadataObj
+							.iterator();
+
+					while (jsonIterator.hasNext()) {
+						final String key = jsonIterator.next().getKey();
+
+						final EmoteMetadata emoteMetadata = metadataObj
+								.get(key).asObject()
+								.asObject(EmoteMetadata.class);
+
+						// id is always structured as emote|{subreddit_id}|{emote_id}
+						// for subreddit emotes
+						if (emoteMetadata.id.split("\\|")[0].equalsIgnoreCase("emote")) {
+							final String emoteId = emoteMetadata.id.split("\\|")[2];
+
+							final String emotePlaceholder = String.format(Locale.getDefault(),
+									":%s:", emoteId);
+
+							final String imgTag = String.format(Locale.getDefault(),
+									"<img src=\"%s\" title=\"%s\"></img>",
+									emoteMetadata.s.u,
+									emotePlaceholder);
+
+							comment.body_html = comment.body_html.replace(emotePlaceholder, imgTag);
```

Comment:
This seems like a bit of an inefficient way to do this -- ideally this would be part of the parser itself. This is probably good enough though, since comments aren't typically that long and most comments don't contain any emotes.

Reply:
So you're saying that this logic should be part of the JSON parser that deserializes the JSON into the reddit comment object? Or are you referring to the `HTMLReader` parser?

Reply:
I think it would fit best into the HTML parser, maybe as a "post processing" step (walk the element tree and split apart text elements which contain emotes). I think the current way is probably fine though.